### ISSUE TITLE
rbx 2.3.0 waits for few seconds before exiting on FreeBSD

### ISSUE THREAD

**Issue by wahjava** [rubinius/rubinius#3204]
rbx is compiled with following `configure` arguments:

```
-P "/usr/local"  -I "/usr/local/include/rubinius"  -L "/usr/local/lib/ruby/rubinius"  -G "/usr/local/lib/ruby/gems/rubinius"  -B "/usr/local/bin"  -M "/usr/local/man"  --cc clang  --cxx clang++  --no-bin-links  --llvm-config /usr/local/bin/llvm-config35  --with-fibers --with-dtrace --release-build  --sitedir "/usr/local/lib/ruby/site_ruby/rubinius"  --vendordir "/usr/local/lib/ruby/vendor_ruby/rubinius"  --stagingdir "/home/abbe/ports/lang/rubinius/work/stage/usr/local" --with-execinfo  --system-name FreeBSD
```

Following is the an excerpt from the output of `echo |truss rbx` while it's waiting:

```
open("/lib/libgcc_s.so.1",O_CLOEXEC,023)         = 9 (0x9)
fstat(9,{ mode=-r--r--r-- ,inode=16400,size=56384,blksize=56832 }) = 0 (0x0)
close(9)                                         = 0 (0x0)
madvise(0x807816000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x80781b000,0x8000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x80780e000,0x7000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x807823000,0x8000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x80782b000,0x10000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x807806000,0x8000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x807815000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x807817000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
_umtx_op(0x805123c00,0x2,0x1885a,0x0,0x0,0x15a5aa0) = 0 (0x0)
_umtx_op(0x80155e080,0xf,0x0,0x0,0x0,0x0)        = 0 (0x0)
_umtx_op(0x803806658,0x15,0x1,0x0,0x0,0x18cb1)   = 0 (0x0)
madvise(0x806416000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x80640e000,0x7000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x806406000,0x8000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x806415000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
_umtx_op(0x80511ec00,0x2,0x18877,0x0,0x0,0x18cb1) = 0 (0x0)
kevent(8,{0x0,EVFILT_TIMER,EV_ADD|EV_ENABLE,0,0x1,0x805034530},1,{},1,{0.001500000 }) = 0 (0x0)

WAITING

kevent(8,{0x0,EVFILT_TIMER,EV_ADD|EV_ENABLE,0,0x2710,0x0},1,{0x0,EVFILT_TIMER,EV_CLEAR,0,0x1,0x805034530},1,0x0) = 1 (0x1)
kevent(8,{0x0,EVFILT_TIMER,EV_DELETE,0,0x0,0x805034530},1,0x0,0,{0.001500000 }) = 0 (0x0)
madvise(0x806016000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x80600e000,0x7000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x806006000,0x8000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
madvise(0x806015000,0x1000,0x5,0xaaaaaaaaaaaaaaab,0x0,0x8030e7be0) = 0 (0x0)
_umtx_op(0x80511e800,0x2,0x18868,0x0,0x0,0x805034530) = 0 (0x0)
kevent(7,{0x4,EVFILT_VNODE,EV_ADD|EV_ENABLE|EV_CLEAR,2,0x0,0x0},1,{0x4,EVFILT_VNODE,EV_CLEAR,6,0x0,0x0},1,0x0) = 1 (0x1)
```

This never happened with `2.2.10`. I'm able to reproduce this on a `10.1-RC4` and `9.3-RELEASE` hosts on `amd64` architecture.

Let me know if there is anything you like me to try, or need more information.

HTH

**Comment by wahjava** [comment]
I've tried `2.5.2`, and the problem is not happening there anymore.

Thanks!

**Comment by brixen** [comment]
@wahjava pretty sure you were seeing the issue fixed in 6d1c999b9f63d12cf5148f3fda2deaab2378754b.
